### PR TITLE
Adds Job resource support. Fixes #30

### LINKED ISSUE #30: Support validation of Jobs
Enable admission controller validation for Job objects.

For more information on Jobs, see https://kubernetes.io/docs/concepts/workloads/controllers/job/.

### PR DESCRIPTION
**Pull request by zhill**

### CODE REVIEW

**Comment by zhill** [comment]
Same basic code as #25 , but better commits and branch name now that we have granular issues to track.

File: cmd/kubernetes-admission-controller/extractor/from_job.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	"k8s.io/api/admission/v1beta1"
+	batchV1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Enforcing compliance with the Extractor type
+var _ Extractor = fromJob
+
+// fromDeployment returns the extracted object metadata and included v1.PodSpecs from the Deployment admission request.
```
Comment (luhring) [inline cmd/kubernetes-admission-controller/extractor/from_job.go:15]: (I'm sorry, @zhill — I missed this in the first PR, but I should've caught it then.) The doc comment needs to match the function name in order to be useful in the Go tooling. In this case, to make the meaning clear, we'd also want to update mentions of "Deployments" elsewhere in the comment, too.

Here's what I'd suggest:

```suggestion
// fromJob returns the extracted object metadata and included v1.PodSpecs from the Job admission request.
```

File: cmd/kubernetes-admission-controller/extractor/from_job_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	// assert
+	assert.NoError(t, err)
+	assert.EqualValues(t, testDeploymentObjectMeta, actualMeta)
+	assert.Contains(t, actualPodSpecs, testPodSpec)
+}
+
+func mockAdmissionRequestFromJob(t *testing.T, job batchV1.Job) v1beta1.AdmissionRequest {
+	t.Helper()
+
+	return mockAdmissionRequestFromObject(t, "Deployment", "deployments", job)
```
Comment (luhring) [inline cmd/kubernetes-admission-controller/extractor/from_job_test.go:34]: Gah, I missed this, too. 😭  This should be specifying "job" values for the resource kind.

**Review by luhring — APPROVED** [review]
@zhill I missed a couple of findings in my first review. I apologize. I can address these myself later if you want to move ahead.

File: cmd/kubernetes-admission-controller/extractor/from_job_test.go (hunk 2 above)
Comment (luhring) [reply cmd/kubernetes-admission-controller/extractor/from_job_test.go:34]: Btw, I'm taking this as feedback that it needs to be easier to mock resources in the test code. I'll strive to improve that in future PRs.

**Pull request merged**